src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/ProjectSystem/RazorProjectInfoEndpointPublisher.cs

```diff
@@ -191,4 +188,8 @@ private void ImmediatePublish(ProjectKey projectKey, string? encodedProjectInfo,
                 parameter,
                 cancellationToken).Forget();
     }
+
+    // Used by tests
+    protected Task WaitUntilCurrentBatchCompletesAsync()
+        => _workQueue.WaitUntilCurrentBatchCompletesAsync();
```

Comment:
A test accessor is the right way to do this one too.

Reply:
Done